libs/api/domains/financial-aid/src/lib/financialAid.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common'
+import type { Auth } from '@island.is/auth-nest-tools'
+import { AuthMiddleware } from '@island.is/auth-nest-tools'
+import { ApplicationApi } from '../../gen/fetch/apis/ApplicationApi'
+
+@Injectable()
+export class FinancialAidService {
+  constructor(private applicationApi: ApplicationApi) {}
+
+  applicationApiWithAuth(auth: Auth) {
+    return this.applicationApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  async hasUserApplicationForCurrentPeriod(auth: Auth, nationalId: string) {
+    console.log('auth', auth)
+    return await this.applicationApiWithAuth(
+      auth,
+    ).applicationControllerGetCurrentApplication({
```

Comment:
What does this API return if there is no valid application? null? If it returns 404, you might want to catch it and return null instead.

Reply:
Ah. Good question. The API Design Guide says it should return 404 if there’s no data to be returned.

That also fits better when documenting the API with OpenAPI, to always have a consistent DTO or a different response status.

Reply:
Nice, will throw 404 😄